File: example-java/src/main/java/com/github/simonpercic/oklogexamplejava/DataFetcher3Component.java
package com.github.simonpercic.oklogexamplejava;

import com.github.simonpercic.oklogexamplejava.data.api.okhttp3.DataModule3;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
@Singleton
@Component(modules = DataModule3.class)
public interface DataFetcher3Component {

    void inject(DataFetcher3 dataFetcher3);
}

File: example-kotlin/src/main/kotlin/com/github/simonpercic/oklogexamplekotlin/DataFetcher3Component.kt
package com.github.simonpercic.oklogexamplekotlin

import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp3.DataModule3

import javax.inject.Singleton

import dagger.Component

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
@Singleton
@Component(modules = arrayOf(DataModule3::class))
interface DataFetcher3Component {

    fun inject(dataFetcher3: DataFetcher3)
}
